File: EasyLearning-BackEnd/easylearning/src/main/java/com/hutech/easylearning/entity/ShoppingCartItem.java
package com.hutech.easylearning.entity;


import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "shopping_cart_item")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ShoppingCartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name ="id", columnDefinition = "VARCHAR(36)")
    String id;

    @Column(name = "shopping_cart_item_name")
    String cartItemName;

    @Column(name = "shopping_cart_item_quantity")
    Integer quantity;

    @Column(name = "shopping_cart_item_imageUrl")
    String imageUrl;

    @Column(name = "shopping_cart_item_price", precision = 10, scale = 3)
    BigDecimal cartItemPrice;

    @Column(name = "shopping_cart_item_price_discount", precision = 10, scale = 3)
    BigDecimal cartItemPriceDiscount;

    @Column(name ="shopping_cart_id")
    String shoppingCartId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shopping_cart_id", referencedColumnName = "id", insertable = false, updatable = false)
    @JsonBackReference
    ShoppingCart shoppingCart;

    @Column(name ="courses_id")
    String courseId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "courses_id", referencedColumnName = "id", insertable = false, updatable = false)
    @JsonBackReference
    Course course;

    @Column(name = "date_create")
    LocalDateTime dateCreate;

    @Column(name = "date_change")
    LocalDateTime dateChange;

    @Column(name = "changed_by")
    String changedBy;

    @Column(name = "is_deleted")
    boolean isDeleted;

    @Version
    private int version;
}

File: EasyLearning-Mobile/flutter_application_elearning/lib/src/Model/shoppingCartItem.dart
class ShoppingCartItem {
  final String id;
  final String cartItemName;
  final String imageUrl;
  final double cartItemPrice;
  final double cartItemPriceDiscount;
  final String shoppingCartId;
  final String courseId;
  final String dateCreate;
  final String dateChange;
  final String changedBy;
  final bool deleted;
  final bool notRegistrable;

  // Constructor
  ShoppingCartItem({
    required this.id,
    required this.cartItemName,
    required this.imageUrl,
    required this.cartItemPrice,
    required this.cartItemPriceDiscount,
    required this.shoppingCartId,
    required this.courseId,
    required this.dateCreate,
    required this.dateChange,
    required this.changedBy,
    required this.deleted,
    required this.notRegistrable,
  });

  // Factory method to create ShoppingCartItem from JSON
  factory ShoppingCartItem.fromJson(Map<String, dynamic> json) {
    return ShoppingCartItem(
      id: json['id'],
      cartItemName: json['cartItemName'],
      imageUrl: json['imageUrl'],
      cartItemPrice: json['cartItemPrice'].toDouble(),
      cartItemPriceDiscount: json['cartItemPriceDiscount'].toDouble(),
      shoppingCartId: json['shoppingCartId'],
      courseId: json['courseId'],
      dateCreate: json['dateCreate'],
      dateChange: json['dateChange'],
      changedBy: json['changedBy'],
      deleted: json['deleted'],
      notRegistrable: json['notRegistrable'],
    );
  }

  // Method to convert ShoppingCartItem to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'cartItemName': cartItemName,
      'imageUrl': imageUrl,
      'cartItemPrice': cartItemPrice,
      'cartItemPriceDiscount': cartItemPriceDiscount,
      'shoppingCartId': shoppingCartId,
      'courseId': courseId,
      'dateCreate': dateCreate,
      'dateChange': dateChange,
      'changedBy': changedBy,
      'deleted': deleted,
      'notRegistrable': notRegistrable,
    };
  }
}
